torch/distributed/checkpoint/filesystem.py

```diff
@@ -75,6 +79,160 @@ def _result_from_write_item(
     )
 
 
+class _TensorLoader(ABC):
```

Comment:
Maybe not in one PR, can we add some comments to these class and functions?

Reply:
Will update comments and docstring in following PRs.